SANBOXED_BUILDS.md

```diff
@@ -0,0 +1,119 @@
+# Sandboxed builds
+
+For convenience, OpenAssetIO provides two different sandboxed build
+environments.
+
+## Docker
+
+A simple way to perform a from-source build is to use the
+[OpenAssetIO-build docker image.](https://github.com/OpenAssetIO/OpenAssetIO/pkgs/container/openassetio-build)
+This image contains all the dependencies required for building and
+testing OpenAssetIO, and functions as a fully configured build
+environment.
+
+> **Note**
```

Comment:
[consider][minor] I'm not convinced this needs to be called out as a `Note`, I worry overuse dilutes the impact of other `Note`s.

Reply:
The notes suit my style of prose (which is to heavily use bracketed information as rambling asides), so they fit in my brain well. I'll remove them though if you think I'm overdoing them, will have to slightly reword the text to fit into the document flow.

Reply:
[a5faf66](https://github.com/OpenAssetIO/OpenAssetIO/pull/735/commits/a5faf66564e38016244744e7ef5a45f72232a400)

Reply:
Docker Hub, even better.